forte/data/ontology/top.py

```diff
@@ -686,5 +690,276 @@ def get(
+
+
+class Box(Region):
+    """ """
+
+    def __init__(
+        self,
+        pack: PackType,
+        image_payload_idx: int,
+        cy: int,
+        cx: int,
+        height: int,
+        width: int,
+    ):
+        # assume Box is associated with Grids
+        super().__init__(pack, image_payload_idx)
+        # center location
+        self._cy = cy
+        self._cx = cx
+        self._height = height
+        self._width = width
+
+    @property
+    def center(self):
+        return (self._cy, self._cx)
+
+    @property
+    def corners(self):
+        """
+        Get corners of box.
+        """
+        return [
+            (self._cy + h_offset, self._cx + w_offset)
+            for h_offset in [-0.5 * self._height, 0.5 * self._height]
+            for w_offset in [-0.5 * self._width, 0.5 * self._width]
+        ]
+
+    @property
+    def box_min_x(self):
+        return max(self._cx - round(0.5 * self._width), 0)
+
+    @property
+    def box_max_x(self):
+        return min(self._cx + round(0.5 * self._width), self.max_x)
+
+    @property
+    def box_min_y(self):
+        return max(self._cy - round(0.5 * self._height), 0)
+
+    @property
+    def box_max_y(self):
+        return min(self._cy + round(0.5 * self._height), self.max_y)
+
+    def is_overlapped(self, other):
+        # If one box is on left side of other
+        if self.box_min_x > other.box_max_x or other.box_min_x > self.box_max_x:
+            return False
+
+        # If one box is above other
+        if self.box_min_y > other.box_max_y or other.box_min_y > self.box_max_y:
+            return False
+        return True
+
+
+class BoundingBox(Box):
+    # logic: grids -> grids cell center (point) -> box metrics
+    def __init__(
+        self,
+        pack: PackType,
+        image_payload_idx: int,
+        height: int,
+        width: int,
+        grid_height: int,
+        grid_width: int,
+        grid_cell_h_idx: int,
+        grid_cell_w_idx: int,
+    ):
+        """
+
+
+        Args:
+            pack (PackType): The container that this BoundingBox will
+                be added to.
+            image_payload_idx (int): the index of the image payload.
+            height (int): the height of the bounding box.
+            width (int): the width of the bounding box.
+            grid_height: the height of the associated grids.
+            grid_width (int): the width of the associated grids.
+            grid_cell_h_idx (int): the height index of the associated grid cell.
+            grid_cell_w_idx (int): the width index of the associated grid cell.
+
+        """
+
+        self.grids = Grids(pack, grid_height, grid_width, image_payload_idx)
+        cy, cx = self.grids.get_grid_cell_center(
+            grid_cell_h_idx, grid_cell_w_idx
+        )
+        super().__init__(pack, image_payload_idx, cy, cx, height, width)
+
+
 SinglePackEntries = (Link, Group, Annotation, Generics, AudioAnnotation)
```

Comment:
images should be placed in the singlepackentries

Reply:
placed it.